frontend/src/components/AnnouncementsList/AnnouncementsList.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react';
+import { useQuery } from 'react-query';
+import Announcement from '../Announcement/Announcement';
+import PropTypes from 'prop-types';
+
+const AnnouncementsList = ({ boardsIds, announcementsNumber, showBoardNames }) => {
+  const { data, isLoading, isError } = useQuery('boardsList', async () => {
+    return await fetch('/boards').then((response) => response.json());
+  });
+
+  const [announcements, setAnnouncemensts] = useState([]);
+  const [windowSize, setWindowSize] = useState('');
+
+  useEffect(() => {
+    const resizeHandler = () => setWindowSize(window.innerWidth);
+    window.addEventListener('resize', resizeHandler);
+
+    return () => {
+      window.removeEventListener(resizeHandler);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!!data && !!data.boards) {
+      console.log(data.boards.filter(({ boardId }) => boardsIds.includes(boardId)));
+      const announcementsToShow = data.boards
+        .filter(({ boardId }) => boardsIds.includes(boardId))
+        .reduce((acc, { announcements }) => {
+          acc.push(...announcements);
+          return acc;
+        }, [])
+        .sort((a, b) => b.date - a.date)
+        .slice(0, announcementsNumber);
+      console.log(announcementsToShow);
+      setAnnouncemensts(announcementsToShow);
+    }
+  }, [data]);
+
+  if (isLoading) return <div>Loading...</div>;
+  if (isError) return <div>Error</div>;
+
+  return (
+    <div data-testid={'announcementsList'}>
```

Comment:
Dlaczego test-id. 
Raczej nie powinniśmy dodawać. 
Jest to do czegokolwiek niezbedne?
Ja przez 3 m-c pisania testów w JEST nie użyłem ani jednego test-id

Zerknij na to (priorytety) - ID jest na samym końcu — ostateczną ostatecznością do użycia ;):
https://testing-library.com/docs/queries/about/#priority

Reply:
Ale jak to inaczej zrobić, jak Announcement jest divem i zawiera divy, a chcemy go przetestować? Wyciągać po divie?

Reply:
usunięte.